### PR TITLE
[suggest] Add command for GM to learn all the available skills at current level

### PR DESCRIPTION
**Pull request by Bogir**
## Changes Proposed:
-  learn all the available skills that can be learned throw ClassMaster at current level

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
Probably better to do after https://github.com/azerothcore/azerothcore-wotlk/pull/6109

**Comment by FrancescoBorzi** [comment]
build fails :(

**Comment by Bogir** [comment]
> 
> 
> build fails :(

изображение
0 errors
Win 7 x64 
VS16 2019

What can it be?

**Comment by Kitzunu** [comment]
```
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/scripts/Commands/cs_learn.cpp: In static member function ‘static void learn_commandscript::learn(Player*, std::unordered_map<unsigned int, CreatureTemplate>::const_iterator, CreatureTemplate)’:
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/scripts/Commands/cs_learn.cpp:137:81: error: unused parameter ‘itr’ [-Werror=unused-parameter]
  137 |     static void learn(Player* player, CreatureTemplateContainer::const_iterator itr, CreatureTemplate cInfo)
      |                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
compilation terminated due to -Wfatal-errors.
```

**Comment by Bogir** [comment]
ok, rebuilding from every begining .... Let's see..

**Comment by Kitzunu** [comment]
It wont show on MSVC because it's a forgiving compiler, but you can see the CI's that look for more stuff

**Comment by Bogir** [comment]
> It wont show on MSVC because it's a forgiving compiler, but you can see the CI's that look for more stuff

ok. What must I do to fix?
Everything in fine for me...
изображение

**Comment by Krejza9** [comment]
@Bogir Install Linux and compile it  (and Stop using Windows for server apps :-D)

**Comment by Bogir** [comment]
> @Bogir Install Linux and compile it (and Stop using Windows for server apps :-D)

I stopped using *nix in 1992 and I will never come back to it.
If your compiller can't build core - this does not prevent us from working under Win :D
You can fix it to work even on Mars if is it necessary for your system or for your ambitions. 
This patch only suggestion and can be not merged, but can be used in private case

File: src/server/scripts/Commands/cs_learn.cpp [hunk 1]
```diff
@@ -105,6 +106,70 @@ class learn_commandscript : public CommandScript, public PlayerCommand
         return true;
     }
 
+
```
Comment (Winfidonarleyan) [inline src/server/scripts/Commands/cs_learn.cpp:109]: Много новых строк, удали эту

**Comment by Krejza9** [comment]
If you program something for AC , You have to program it so that it can be compiled on Linux, Windows, MacOS.
If you don't care if it doesn't work on other systems.
Then ask someone to fix it, or cancel the PR

**Comment by Bogir** [comment]
done

File: src/server/scripts/Commands/cs_learn.cpp [hunk 2]
```diff
@@ -30,6 +30,7 @@ class learn_commandscript : public CommandScript, public PlayerCommand
     {
         static std::vector<ChatCommand> learnAllMyCommandTable =
         {
+            { "lvl",            SEC_GAMEMASTER,  false, &HandleLearnAllMyLvlCommand,         "" },
```
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_learn.cpp:33]: alphabetical order

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/scripts/Commands/cs_learn.cpp [hunk 3]
```diff
@@ -105,6 +106,70 @@ class learn_commandscript : public CommandScript, public PlayerCommand
         return true;
     }
 
+    static bool HandleLearnAllMyLvlCommand(ChatHandler* handler, char const* /*args*/)
```
Comment (Yehonal) [inline src/server/scripts/Commands/cs_learn.cpp:109]: Would be better to move the logic of this function in a separate file in order to let people reuse it for other purposes (custom scripts etc).

In this way, modders could create scripts that automatically teach you spells when you level up for instance.

**Review by Yehonal — CHANGES_REQUESTED** [review]

File: src/server/scripts/Commands/cs_learn.cpp (hunk 3 above)
Comment (Kitzunu) [reply src/server/scripts/Commands/cs_learn.cpp:109]: I don't agree with handling commands in other files than the designated ones

Comment (Yehonal) [reply src/server/scripts/Commands/cs_learn.cpp:109]: I said "the logic", which means creating a new function elsewhere and using that one in `HandleLearnAllMyLvlCommand`

that function can be reused then in other scripts/modules/logics

Comment (Kitzunu) [reply src/server/scripts/Commands/cs_learn.cpp:109]: ah, I oversaw that :D

**Pull request closed without merging**